### PR TITLE
arm64: handling of system registers added in ARMv8.1/2

### PR DESCRIPTION
**Pull request by semihalf-oleksy-michalina**

### CODE REVIEW

File: arch/AArch64/AArch64BaseInfo.h [hunk 1]
```diff
@@ -334,6 +334,7 @@ enum A64SysRegROValues {
 	A64SysReg_ID_MMFR1_EL1      = 0xc00d, // 11  000  0000  0001  101
 	A64SysReg_ID_MMFR2_EL1      = 0xc00e, // 11  000  0000  0001  110
 	A64SysReg_ID_MMFR3_EL1      = 0xc00f, // 11  000  0000  0001  111
+  A64SysReg_ID_MMFR4_EL1      = 0xc016, // 11  000  0000  0010  110
```
Comment (aquynh) [inline arch/AArch64/AArch64BaseInfo.h]: please use tabs for indentation here, and also in other places.

Comment (semihalf-oleksy-michalina) [inline arch/AArch64/AArch64BaseInfo.h]: I hope it's okay now.

**Comment by semihalf-oleksy-michalina** [comment]
I guess that would be all when it comes to new system registers. 
(I didn't add them previously as they weren't listed in the same place as the rest.)

**Comment by semihalf-oleksy-michalina** [comment]
Sorry for the trouble, I hope it's okay now. Let me know if it's not.

**Comment by aquynh** [comment]
on these new added register stuffs, did you look at the related latest LLVM code?

**Comment by semihalf-oleksy-michalina** [comment]
I added the changes only using info in ARM manuals. But when I was looking around to find out what's missing I noticed that some changes in LLVM (~2015) mention the missing registers. Should I take a look at the newest ones?
What exactly do you have in mind?

**Comment by semihalf-oleksy-michalina** [comment]
Is there something wrong with the changes I've made?

**Comment by aquynh** [comment]
merged now, but we still need to look at the latest LLVM code to adapt to upstream.

thanks!

**Pull request merged**